batch/batch/front_end/front_end.py

```diff
@@ -672,6 +719,68 @@ async def get_batches_v2(request, userdata):  # pylint: disable=unused-argument
     return json_response({'batches': batches})
 
 
+async def _query_job_groups(request, batch_id: int, job_group_id: int, last_child_job_group_id: Optional[int]):
+    db: Database = request.app['db']
+
+    record = await db.select_and_fetchone(
+        """
+SELECT 1
+FROM job_groups
+LEFT JOIN batches ON batches.id = job_groups.batch_id
+LEFT JOIN batch_updates
+  ON job_groups.batch_id = batch_updates.batch_id AND job_groups.update_id = batch_updates.update_id
+WHERE job_groups.batch_id = %s AND job_groups.job_group_id = %s AND NOT deleted AND (batch_updates.committed OR job_groups.job_group_id = %s);
+""",
+        (batch_id, job_group_id, ROOT_JOB_GROUP_ID),
+    )
+    if not record:
+        raise NonExistentJobGroupError(batch_id, job_group_id)
+
+    sql, sql_args = parse_list_job_groups_query_v1(batch_id, job_group_id, last_child_job_group_id)
+    job_groups = [job_group_record_to_dict(record) async for record in db.select_and_fetchall(sql, sql_args)]
```

Comment:
This isn't copacetic. The first query could return a non-deleted batch and then this query could list the jobs in a deleted batch (which, as I noted below, returns an empty list).

Either we need a transaction or we need to use one query. My inclination is to prefer one round trip to the database whenever possible, but I can't think of a simple way to learn all three of the following in one query:
1. The batch exists and is not deleted.
2. The job group exists and is not deleted.
3. The list of job groups in the job group.

Let's use a transaction.